test/host_tests/sal_tcpserver.py

```diff
@@ -0,0 +1,152 @@
+                
+        finally:
+            self._shutdown_request = False
+        
+        
+
+class TCPHandler(BaseRequestHandler):
+    MAX_RX_SIZE = 8192
+    def handle(self):
+        """ One handle per connection
+        """
+        print("HOST: Connection received...\n")
+        while True:
+            try:
+                data = self.request.recv(self.MAX_RX_SIZE)
+            except Exception as e:
+                    print("HOST: detected unexpected exception: %s" % str(e))
+                    break
+                
+            if not data: 
+                break
+            
+            print("HOST: Received %s bytes of data\n" % len(data))
+            
+            # try not shutting down the connection and wait for the remote end to close it
+            if data == "shutdown":
+                print("HOST: Requesting server shutdown\n")
+                # give the remote end time to close its connection, which is part of the test
+                time.sleep(5)
+                self.server._shutdown_request = True
+                break
+
+            self.request.sendall(data)
+            
+
+class SalTcpServerTest(BaseHostTest):
+    ERR_SUCCESS = "success"
+    name = 'sal_tcpserver'
+
+    def send_server_ip_port(self, selftest, ip_address, port):
+        """ Set up network host. Reset target and and send server IP via 
+        serial to a mbed board."""
+
+        self.watchdog = 0
+
+        # Read 3 lines which are sent from client, This will be fixed in greentea
+        for i in range(0, 3):
+            c = selftest.mbed.serial_readline()
+            if c is None:
+                selftest.print_result(self.RESULT_IO_SERIAL)
+                return
+            selftest.notify("MBED: " + c.strip())
+
+
+        selftest.notify("HOST: Sending server IP Address to target...")
+        msg = ip_address + ":" + str(port) + "\n"
+        selftest.mbed.serial_write(msg)
+        
+        # mbed greentea framwork support the following for outputing serial console emitted 
+        # from the target, but it doesnt work relibly for me.
+        # selftest.dump_serial()
+        # selftest.dump_serial_end()
+        
+        # until its fixed, the following emits the serial console trace
+        while True:
+            c = selftest.mbed.serial_readline()
+            if c is None:
+                selftest.print_result(self.RESULT_IO_SERIAL)
+                return
+            
+            print "MBED: " + c.strip()
+            
+            # look for the end tag in serial output denoting the test has 
+            # finished, and this can return.
+            if c.strip() == "{{end}}":
+                print "HOST: Terminating Test"
+                break
+
+            # null lines are periodically generated, which can be used to trigger the watchdog
+            elif c.strip() == "":
+                self.watchdog += 1
+                if self.watchdog > 10:
```

Comment:
This watchdog is 10, but the previous is 20.0. Could these be defined at the top of the file, or made configurable?